src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import { formatString } from '../servises/formatString';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  people: Person[];
+  onSelect: (person: Person | null) => void;
+};
+
+export const Autocomplete: FC<Props> = ({ people, onSelect }) => {
+  const [query, setQuery] = useState('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useMemo(() => {
+    return debounce(setAppliedQuery, 300);
+  }, []);
+
+  useEffect(() => {
+    return () => applyQuery.cancel();
+  }, [applyQuery]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
```

Comment:
Ensure that the `applyQuery` function does not get called if the input is spaces only. This is a requirement from the checklist: "make sure that filter won't be called if user entered spaces only."